tests/30rooms/01state.pl

```diff
@@ -242,14 +242,24 @@ sub get_test_state
 {
    my ( $user, $room_id, $type ) = @_;
 
-   matrix_initialsync_room( $user, $room_id, limit => 0 )->then( sub {
-      my ( $body ) = @_;
+   matrix_sync( $user,
+      filter            => '{"room":{"rooms":["'.$room_id.'"]},"account_data":{},"presence":{"types":[]}}',
+      update_next_batch => 0,
+      set_presence      => "offline",
+   )->then( sub {
+        my ( $body ) = @_;
 
-      my %state_by_type = partition_by { $_->{type} } @{ $body->{state} };
+        # We expect the event to be in the initial sync
+        sync_timeline_contains( $body, $room_id, sub {
```

Comment:
I think `sync_timeline_contains` returns a truthy result? shouldn't we be doing something with that result (otherwise the call is redundant?)

Reply:
I just copy-pasta'd from elsewhere, but I figured that if the timeline didn't hit the `sub` and match the event we want, that the `Future` wouldn't be done?